File: app/src/main/java/com/rhionin/brandonsanderson/WorkInProgress.java
package com.rhionin.brandonsanderson;

/**
 *
 */
public class WorkInProgress {

    private String title;
    private Integer progress;
    private Integer prevProgress;

    public WorkInProgress() {
    }

    public WorkInProgress(String title, int progress) {
        this.title = title;
        this.progress = progress;
    }

    public Integer getProgress() {
        return progress;
    }

    public void setProgress(Integer progress) {
        this.progress = progress;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Integer getPrevProgress() {
        return prevProgress;
    }

    public void setPrevProgress(Integer prevProgress) {
        this.prevProgress = prevProgress;
    }
}

File: lib/work_in_progress.dart
import "dart:convert";

class WorkInProgress {
  String title;
  int progress;
  int prevProgress;

  WorkInProgress({this.title = "", this.progress = 0, this.prevProgress = 0});

  WorkInProgress.fromJson(Map<String, dynamic> json)
      : title = json['title'] ?? '',
        progress = json['progress'] ?? 0,
        prevProgress = json['prevProgress'] ?? 0;

  WorkInProgress.fromJsonString(String jsonStr)
      : this.fromJson(jsonDecode(jsonStr));

  Map<String, dynamic> toJson() =>
  {
    'title': title,
    'progress': progress,
    'prevProgress': prevProgress,
  };

  int getProgress() {
    return progress;
  }

  void setProgress(int progress) {
    this.progress = progress;
  }

  String getTitle() {
    return title;
  }

  void setTitle(String title) {
    this.title = title;
  }

  int getPrevProgress() {
    return prevProgress;
  }

  void setPrevProgress(intprevProgress) {
    this.prevProgress = prevProgress;
  }
}
